File: ++/two.cpp
#include <iostream>
using namespace std;
void drink(int N, int M) {
    int vipito = N;
    int step = 1;
    int pusto = N;
    cout << "input: " << N << " " << M << "\nДано: ";
    for (int i = 0; i < N; ++i) cout << "😊";
    cout << endl;
    cout << step << " Шаг ";
    for (int i = 0; i < pusto; ++i) cout << "😞";
    cout << endl;
    while (pusto >= M) {
        step++;
        int smen = pusto / M;
        pusto = pusto % M + smen;
        vipito += smen;

        cout << step << " Шаг ";
        for (int i = 0; i < pusto; ++i) cout << "😊";
        cout << endl;

        step++;
        cout << step << " Шаг ";
        for (int i = 0; i < pusto; ++i) cout << "😞";
        cout << endl;
    }
    cout << "Output: " << vipito << " " << step << endl;
}
int main() {
    int N, M;
    cout << "Введите количество банок: ";
    cin >> N;
    cout << "Количество банок для обмена: ";
    cin >> M;
    drink(N, M);
    return 0;
}

File: ruby/two.rb
def banki(n, m)
    pusto = n 
    step = 1
    vsego_vipito = n
    puts"Input: " + n.to_s + "," + m.to_s
    puts "Дано: " + "😊" * n
    puts step.to_s + "Шаг" + "😞" * n
    while pusto >= m 
        step += 1
        smen = pusto / m 
        pusto = pusto % m + smen
        vsego_vipito += smen
        puts(step.to_s + "Шаг" + "😊" * pusto)
        step += 1
        puts(step.to_s + "Шаг" + "😞" * pusto)
    end
    puts "Output: " + vsego_vipito.to_s + "," + step.to_s
    return vsego_vipito, step
end
print("Введите количество банок: ")
n = gets.chomp().to_i
print("Количество банок для обмена: ")
m = gets.chomp().to_i
banki(n, m)
